src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,963 @@
 };
 
-u16 D_8082B5E0[] = {
-    0xAF, 0xB3, 0xAA, 0xB1, 0xA9, 0xB2, 0xA8, 0xB0, 0xAC, 0xAE,
+void KaleidoScope_DrawWorldMap(PlayState* play) {
+    s16 sceneId;
+    s16 t;
+    s16 n;
+    s16 j;
+    s16 k;
+    s16 i;
+    PauseContext* pauseCtx = &play->pauseCtx;
+    s16 rectLeft;
+    s16 rectRight;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    KaleidoScope_SetCursorVtx(pauseCtx, pauseCtx->cursorSlot[PAUSE_MAP] * 4, pauseCtx->mapPageVtx);
+
+    // Draw the world map image
+    if ((pauseCtx->pageIndex == PAUSE_MAP) && (pauseCtx->state == 6) &&
+        ((pauseCtx->unk_200 == 0) || (pauseCtx->unk_200 == 3)) && YREG(6) && (pauseCtx->state != 7) &&
+        !((pauseCtx->state >= 8) && (pauseCtx->state <= 0x12))) {
+
+        // Draw the world map image flat
+        // Because it is flat, the texture is loaded by filling it in 8 rows at a time.
+        // 8 is chosen because it is smaller than `TMEM_SIZE / 2 / WIDTH` and divides the texture's height.
+        // (`TMEM_SIZE / 2` because the texture is color-indexed so the TLUT uses the other half of TMEM.)
+
+        func_8012C628(play->state.gfxCtx);
+
+        gDPSetTextureFilter(POLY_OPA_DISP++, G_TF_POINT);
+        gDPLoadTLUT_pal256(POLY_OPA_DISP++, gWorldMapImageTLUT);
+        gDPSetTextureLUT(POLY_OPA_DISP++, G_TT_RGBA16);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+
+        // Process the 128 rows of pixels for gWorldMapImageTex 8 rows at a time over 16 iterations
+        // Loop over yPos (t), textureIndex (j)
+        for (t = 62, j = 0; j < 16; j++, t += 8) {
+            gDPLoadTextureBlock(POLY_OPA_DISP++, (u8*)gWorldMapImageTex + j * (WORLD_MAP_IMAGE_TEX_WIDTH * 8),
+                                G_IM_FMT_CI, G_IM_SIZ_8b, WORLD_MAP_IMAGE_TEX_WIDTH, 8, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                                G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+            rectLeft = 51 << 2;
+            rectRight = rectLeft + (WORLD_MAP_IMAGE_TEX_WIDTH << 2);
+            gSPTextureRectangle(POLY_OPA_DISP++, rectLeft, t << 2, rectRight, (t << 2) + (8 << 2), G_TX_RENDERTILE, 0,
+                                0, 1 << 10, 1 << 10);
+        }
+
+        func_8012C8AC(play->state.gfxCtx);
+
+    } else {
+        // Draw the world map angled
+        // Because it is at an angle, vertices are used to place it.
+        // The structure of the loops here is to satisfy the constraints of both TMEM and the size of the vertex cache.
+        // - Each loop iteration loads 9 rows, because 9 is the largest number smaller than `TMEM_SIZE / 2 / WIDTH`
+        // - Each loop is at most 8 iterations long because each row uses 4 vertices and the vertex cache has size 32 =
+        // 8 * 4 Hence there is one loop of length 8, one of length 6, and then the remaining `128 - (8 + 6) * 9 = 2`
+        // rows are drawn at the end.
```

Comment:
Formatter messed this one up, the Hence line should not be in with the bullets. Two possible things that might fix it are leaving a blank line after the bullets, or adding more spaces before the `-`s (which I think might be required anyway?).

Reply:
I tried to make it better, looks a bit funny but still better than what it was